exonum/src/helpers/fabric/clap_backend.rs

```diff
@@ -29,9 +29,10 @@ impl ClapBackend {
         I: IntoIterator<Item = T>,
         T: Into<OsString> + Clone,
     {
+        let authors = crate_authors!("\n"); // crate_authors! broken in tx_sender
```

Comment:
I did understand the problem, but I didn't get this comment. Is it useful at all?

Reply:
That might be fixed in a bit different way, just by a minor shift.
It seems that the func chain works if it is closed by get_matches(), like:
```
let matches = clap::App::new("Exonum").author(crate_authors!("\n")).get_matches();
```

So we can change the existing code so:
```
pub fn execute(commands: &[CollectedCommand]) -> Feedback {
    let subcommands: Vec<_> = commands
        .iter()
        .map(|command| ClapBackend::command_into_subcommand(command))
        .collect();

    let matches = clap::App::new("Exonum application based on fabric configuration.")
        .version(crate_version!())
        .author(crate_authors!("\n"))
        .about("Exonum application based on fabric configuration.")
        .subcommands(subcommands.into_iter())
        .get_matches();

    let subcommand = matches.subcommand();
    for command in commands {
        if command.name() == subcommand.0 {
            return command.execute(Context::new_from_args(
                command.args(),
                subcommand.1.expect("Arguments not found."),
            ));
        }
    }

    panic!("Subcommand not found");
}
```

From my perspective that is bug on clap-rs side, since even their example for the macros doesn't work.

Reply:
I like this solution more than creating temporary variable.